File: DSA/ARARTIK_TYAGI_590011633/DAY_4_QUESTION_2.py
# Input: Array and k
nums = list(map(int, input("Enter the array elements (space-separated): ").split()))
k = int(input("Enter the number of steps to rotate: "))

n = len(nums)
k = k % n  # To handle cases where k > n

# Rotating the array
rotated = nums[-k:] + nums[:-k]
print("Rotated array:", rotated)

File: DSA/RATNA_KHANNA_590013141/day4question2.c
#include<stdio.h>
int main()
{
    int n;
    printf("enter number of elements in array\n");
    scanf("%d",&n);
    int nums[n];
    printf("enter array elements\n");
    for(int i=0;i<n;i++)
    {
        scanf("%d",&nums[i]);
    }
    int k;
    printf("enter value of k\n");
    scanf("%d",&k);
    int a[n];
    int c=n-k;
    int j=0;
    for(int i=c;i<n;i++)
    {
        a[j]=nums[i];
        j++;
    }
    for(int i=0;i<c;i++)
    {
        a[j]=nums[i];
        j++;
    }
    
    printf("rotated array\n");
    for(int i=0;i<n;i++)
    {
        printf("%d,",a[i]);
    }

return 0;
}
